src/wms/src/test/java/org/geoserver/wms/capabilities/CapabilitiesXmlReaderTest.java

```diff
@@ -39,5 +41,58 @@ public void testParseXmlGetCapabilities() throws Exception {
         assertEquals("GetCapabilities", request.getRequest());
         assertEquals("1.2.0", request.getVersion());
         assertEquals("1", request.getUpdateSequence());
+        assertNull(request.isNoRootLayer());
+    }
+
+    @Test
+    public void testParseXmlGetCapabilitiesNoRootLayerTrue() throws Exception {
+        CapabilitiesXmlReader reader =
+                new CapabilitiesXmlReader(EntityResolverProvider.RESOLVE_DISABLED_PROVIDER);
+
+        String plainRequest =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" //
+                        + "<ogc:GetCapabilities xmlns:ogc=\"http://www.opengis.net/ows\" " //
+                        + "         xmlns:gml=\"http://www.opengis.net/gml\" " //
+                        + "         version=\"1.2.0\" updateSequence=\"1\" " //
+                        + "         noRootLayer=\"true\" " //
```

Comment:
Seeing here and everywhere else, negations everywhere... would it had been better to have a rootLayer attribute/flag everywhere defaulting to true instead? But I understand it would make for a likely wide change in the PR....